core/charm/repository/charmhub.go

```diff
@@ -65,10 +65,6 @@ func NewCharmHubRepository(logger Logger, chClient CharmHubClient) *CharmHubRepo
 // When charmstore goes, we could potentially rework how the client requests
 // the store.
 func (c *CharmHubRepository) ResolveWithPreferredChannel(charmURL *charm.URL, requestedOrigin corecharm.Origin, macaroons macaroon.Slice) (*charm.URL, corecharm.Origin, []string, error) {
-	if charmURL.Revision != -1 {
```

Comment:
Nice catch. I could have sworn we got rid of this when back-porting the other bits!